DllImportGenerator/DllImportGenerator/MarshallingAttributeInfo.cs

```diff
@@ -10,7 +10,10 @@ namespace Microsoft.Interop
     // for C# 10 discriminated unions. Once discriminated unions are released,
     // these should be updated to be implemented as a discriminated union.
 
-    internal abstract record MarshallingInfo {}
+    internal abstract record MarshallingInfo
+    {
+        public const MarshallingInfo Empty = null!;
```

Comment:
Id rather we make a new NoMarshalInfo record that inherits from MarshallingInfo than explicitly working around nulls like this.

Reply:
I need a constant for the switch statement. I don't see why this matters in this case. The nullability feature being enabled forces this to not compare against `null` in practice. I feel like we are trying to fix the type system in a way that just doesn't work.

Reply:
Ah... I think I think see the suggestion. I think we are just playing more type system games, but I can do that.